src/pdf/ShoppingList/ShoppingListPDF.tsx

```diff
@@ -0,0 +1,235 @@
+        borderStyle: "solid",
+        border: 1,
+        padding: 5,
+    },
+    tableRow: {
+        display: "flex",
+        flexDirection: "row",
+        borderBottomStyle: "solid",
+        borderBottom: 1,
+        paddingVertical: 4,
+        textAlign: "center",
+    },
+    tableItemDescription: {
+        width: "25%",
+    },
+    tableQuantity: {
+        width: "35%",
+    },
+    tableNotes: {
+        width: "30%",
+    },
+    tableDone: {
+        width: "10%",
+    },
+    title: {
+        fontSize: "30pt",
+        fontFamily: "Helvetica-Bold",
+    },
+    subtitle: {
+        fontSize: "24pt",
+        fontFamily: "Helvetica",
+    },
+    keyText: {
+        fontSize: "12pt",
+        fontFamily: "Helvetica-Bold",
+        textTransform: "capitalize",
+    },
+    normalText: {
+        fontSize: "12pt",
+        fontFamily: "Helvetica",
+    },
+    input: {
+        fontSize: "12pt",
+        fontFamily: "Helvetica-Bold",
+        paddingTop: 20,
+    },
+});
+
+const OneLine: React.FC<{ header: string; value: string }> = ({ header, value }) => {
+    return (
+        <Text style={styles.keyText}>
+            {header}: <Text style={styles.normalText}>{value}</Text>
+        </Text>
+    );
+};
+
+const TableHeadings: React.FC<{}> = () => {
+    return (
+        <View style={styles.tableRow}>
+            <View style={styles.tableItemDescription}>
+                <Text style={styles.keyText}>Item Description</Text>
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.keyText}>Quantity</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.keyText}>Notes</Text>
+            </View>
+            <View style={styles.tableDone}>
+                <Text style={styles.keyText}>Done</Text>
+            </View>
+        </View>
+    );
+};
+
+const ItemToRow: React.FC<Item> = (item) => {
+    return (
+        <View style={styles.tableRow}>
+            <View style={styles.tableItemDescription}>
+                <Text style={styles.normalText}>{item.description}</Text>
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.normalText}>{item.quantity}</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.normalText}>{item.notes}</Text>
+            </View>
+            <View style={styles.tableDone}></View>
+        </View>
+    );
+};
+
+const DisplayItemsList: React.FC<{ itemsList: Item[] }> = ({ itemsList }) => {
+    return (
+        <View>
+            {itemsList.map((item, index) => (
+                <ItemToRow {...item} key={index} />
+            ))}
+        </View>
+    );
+};
+
+const formatCamelCaseKey = (objectKey: string): string => {
+    return objectKey.replaceAll(/([a-z])([A-Z])/g, "$1 $2");
+};
+
+const DisplayAsBlock: React.FC<BlockProps> = <Prop extends BlockProps>(data: Prop) => {
```

Comment:
What is the advantage of using extends here as opposed to just putting in a new interface or just writing the type out?

Reply:
`BlockProps` is the union of several interfaces, each with their own keys. And so to prevent type errors in the `data[propKey]` bit from occurring due to keys being specific to only one and not all of the possible interfaces that can be passed in, generics are used. 
(i don't think i explained it well, let me know if you want more explanation)

Reply:
Varun says this could work :) `const DisplayAsBlock: React.FC<BlockProps> = (data) => {`